edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/metrics/ExceptionCounter.cs

```diff
@@ -25,7 +25,10 @@ public ExceptionCounter(Dictionary<Type, string> recognizedExceptions, HashSet<T
                 "exceptions_total",
                 "The number of exceptions thrown of the given type",
                 new List<string> { "exception_name" }));
+        }
 
+        public void Start()
```

Comment:
why do you need to introduce "Start" method?

Reply:
This allows it to only be created and started if metrics are enabled. If metrics aren't enabled, it won't be started and autofac shouldn't create it.